File: src/BusinessReports.WebApp/src/app/dictionaries/indicator/indicator.ts
export class Indicator {
    public id: number;
    public code: string;
    public name: string;
    public indicatorType: IndicatorType;
    public arguments: string;
    public indicatorTypeName: string;

    constructor() {
    }
}

export enum IndicatorType {
    Primitive = 1,
    Formula = 2
}

File: src/BusinessReports.WebApi/Models/Dictionary/Indicator.cs
using Avocado.Core;
using Avocado.WebApi.Models;
using BusinessReports.Entity.Dictionary;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace BusinessReports.WebApi.Models.Dictionary
{
    public class Indicator : IModel, IdAccessor
    {
        public int Id { get; set; }

        [Required]
        [MaxLength(20)]
        public string Code { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        public IndicatorType IndicatorType { get; set; }

        [MaxLength(1000)]
        public string Arguments { get; set; }

    }
}
